### PR TITLE
Centos 6 support

### PR DESCRIPTION
**Pull request by armiller**
This adds support for installing camo on centos 6. For centos we use the runit cookbook instead of upstart because centos does not have the `start-stop-daemon` package. Due to the lack of tests consider this pull request a work in progress. Feedback is welcome.

@svanzoest I pulled in the license header from the default recipe. Does that need to be changed or updated?

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -6,10 +6,23 @@
 # deployment
 default['camo']['app_name'] = 'camo'
 default['camo']['path'] = '/srv/camo'
-default['camo']['user'] = 'www-data'
+
+if platform_family?('debian', 'ubuntu')
```
Comment (svanzoest) [inline attributes/default.rb]: I do not think there is a need for this. It is easy enough to override the user in a wrapper cookbook. We can just set it to `camo`.

File: recipes/default.rb [hunk 2]
```diff
@@ -19,6 +19,10 @@
 include_recipe 'nodejs'
 include_recipe 'git'
 
+if platform_family?('rhel')
+  user node['camo']['user']
```
Comment (svanzoest) [inline recipes/default.rb]: no need to put a platform_family guard around this.

File: recipes/default.rb [hunk 3]
```diff
@@ -47,29 +51,5 @@
...
-service node['camo']['app_name'] do
-  case node['platform']
-  when 'ubuntu'
-    if node['platform_version'].to_f >= 9.10
-      provider Chef::Provider::Service::Upstart
-    end
-  end
-  action [:enable, :start]
-end
+include_recipe 'camo::init'
```
Comment (svanzoest) [inline recipes/default.rb]: Any thoughts on maybe making this similar to install_method? As in
```ruby
include_recipe "camo::#{node['camo']['init_style']}"

**Comment by svanzoest** [comment]
Good stuff. We will want to add a systemd init_style as well as that would be useful for centos 7 and ubuntu 14.10. Thanks for your work on this.

File: attributes/default.rb (hunk 1 above)
Comment (drpebcak) [inline attributes/default.rb]: Also, if you are doing `platform_family?` you only need debian in there - ubuntu will match with that.

File: recipes/default.rb (hunk 3 above)
Comment (armiller) [inline recipes/default.rb]: That sounds much better. I will go ahead and update to that style.

Comment (armiller) [inline recipes/default.rb]: @svanzoest So i'm thinking for the init recipes we use the scheme `_upstart.rb`, `_systemd.rb`, `_runit`, etc. I see this pattern when including a certain type of installation/service in cookbooks. Thoughts?

**Comment by armiller** [comment]
Do we need two different 12.04 kitchen suites? I would like to remove the `canonical-ubuntu-12.04`.

**Comment by svanzoest** [comment]
@armiller I mostly had the canonical version in there to be able to test it next to the Chef Software provided version. I think at this point I am okay with removing it and rely on Chef Software and Canonical to resolve any discrepancies.

**Comment by svanzoest** [comment]
@armiller I have pulled this in and added some tests. I'd love to hear your feedback and ensure that this still works for you.

File: recipes/default.rb (hunk 3 above)
Comment (svanzoest) [inline recipes/default.rb]: Yes, sounds good. I ended up adding an `init` and `install` in there to categorize them a bit.

**Comment by armiller** [comment]
@svanzoest sounds good. I will review it sometime this weekend when I have time.

**Pull request merged**